lib/sendingDomains.js

```diff
@@ -0,0 +1,175 @@
+'use strict';
+
+var request = require('request')
+  , config = require('./configuration');
+
+/**
+ * Private Method for migrating the flat user input into
+ * a format that the Sending Domains REST API expects
+ *
+ * This method does not perform any validation of the user's input, we
+ * rely on the API to return appropriate errors for all the weird combinations
+ * of things, like you can't specify rfc822 content and then specify html/plaintext
+ *
+ * @param input object Flat object of configuration for the sending domain to be sent
+ * @returns object
+ */
+var toApiFormat = function(input) {
+  var model = {
+    dkim: {}
+  };
+
+  model.domain = input.domain;
+  model.dkim['private'] = input.privateKey;
+  model.dkim['public'] = input.publicKey;
+  model.dkim.selector = input.selector;
+  model.dkim.headers = input.headers;
+
+  return model;
+};
+
+/**
+ * Private Method for putting URL construction logic into one place
+ *
+ * @returns string Fully constructed URL for contacting the Sending Domains API
+ */
+var constructURL = function () {
+  return config.options.protocol + '://' + config.options.host + (config.options.port ? ':' + config.options.port: '') + '/api/' + config.options.version + '/sending-domains';
+};
+
+/**
+ *  Private Method for issuing GET request to Sending Domains API
+ *
+ *  This method is responsible for getting the collection _and_
+ *  a specific entity from the Sending Domains API
+ *
+ *  If DomainName parameter is omitted, then we fetch the collection
+ *
+ *  @param DomainName string Domain Name of specific Sending Domain to retrieve
+ *  @param callback function
+ */
+var fetch = function(domainName, callback) {
```

Comment:
we never built a base resource, but now that we have 2, it will probably be a good idea to abstract some common things.  I can create a separate issue for that though